### PR TITLE
Plugins Catalog: Enable install controls based on the pluginAdminEnabled flag

### PR DESCRIPTION
**Pull request by leventebalogh**
**Related issue:** https://github.com/grafana/grafana/issues/40503

### What changed
- [x] stop routing to the old version of the plugin settings
- [x] simplify routing by removing the deprecated routes and creating a routes config specifically for `plugins/admin`
- [x] remove unreferenced old plugins config components
- [x] organise still used components and tests into subfolders

_The clean-up is going to be done in batches (smaller separate PRs). Upcoming ones are cleaning up the plugin state handling and ~removing the feature flag~ changing how the feature-flag works for the plugins catalog._

### CODE REVIEW

File: public/app/features/datasources/state/actions.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { updateNavIndex } from 'app/core/actions';
 import { getBackendSrv } from 'app/core/services/backend_srv';
 import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
 import { importDataSourcePlugin } from 'app/features/plugins/plugin_loader';
-import { getPluginSettings } from 'app/features/plugins/PluginSettingsCache';
+import { getPluginSettings } from 'app/features/plugins/components/PluginSettingsCache';
```
Comment (jackw) [inline public/app/features/datasources/state/actions.ts:8]: nit: is this a component?

**Review by jackw — APPROVED** [review]
Great stuff @leventebalogh  👏 

Took it for a spin and works as expected.

I'm guessing we need to change the feature flag behaviour before we can merge any of these "remove deprecated code" PRs otherwise we'll break main for anyone which has the feature flag set `false`?

**Review by mckn — APPROVED** [review]
LGTM!

**Comment by leventebalogh** [comment]
@jackw 
> I'm guessing we need to change the feature flag behaviour before we can merge any of these "remove deprecated code" PRs otherwise we'll break main for anyone which has the feature flag set false?

Yes, that's right, thanks for pointing it out mate. Will merge the other 2 outstanding PRs into this one before merging the whole if it is fine with you?

File: public/app/features/datasources/state/actions.ts (hunk 1 above)
Comment (leventebalogh) [reply public/app/features/datasources/state/actions.ts:8]: You are right, it's not 🤦‍♂️ 

Probably the name deceived me, haha.

Comment (leventebalogh) [reply public/app/features/datasources/state/actions.ts]: Fixed in https://github.com/grafana/grafana/pull/41686/commits/767b16f237cdbc308a6ee166da312180f98a8c18.

**Pull request merged**